README.md

```diff
@@ -1,2 +1,321 @@
+For example:
+
+```js
+const toast = document.createElement('std-toast');
+console.log(toast.open); // false
+```
+
+#### Contents
+
+The `<std-toast>` element can be used as a generic container,
+but will work best (in terms of default styling and events) if the developer conforms to the following content model:
+
+- The first child node,
+  typically either a text node or a container like `<p>`,
+  provides a message.
+- If more than one child node is present,
+  then the last child element,
+  which can be either an `<a>` or a `<button>`,
+  provides a call to action.
+
+TODO: what about title or icon? They should potentially also be accomodated, in a similar fashion.
+
+Thus, the following would all work well out of the box:
+
+```html
+<std-toast>Hello world!</std-toast>
+
+<std-toast><p>Hello world!</p></std-toast>
+
+<std-toast>
+  <p>Hello world!</p>
+  <button>Click me!</button>
+</std-toast>
+
+<std-toast>
+  <p>Hello world!</p>
+  <a href="https://example.com/" target="_blank">Click me!</button>
+</std-toast>
+```
+
+These can equivalently be created via JavaScript:
+
+```js
+// TODO: write this once we figure out the JS API for supplying HTML messages,
+// and for supplying customizable actions
+```
+
+More complex toasts,
+that don't fit the above content model,
+would require more custom handling on the part of the developer:
+
+```html
+<std-toast>
+  <p>Hello world!</p>
+  <p>A second paragraph?!</p>
+
+  <button>Action 1</button>
+  <button>Action 2</button>
+</std-toast>
+```
+
+Such an unusual toast would still integrate with other toasts in terms of stacking and positioning behavior,
+and some of the default styles that are inherited may be useful.
+But the page author will need to handle the action clicks themselves (instead of using the `"actionclick"` event),
+and will need to add additional styling to handle the extra contents.
+
+TODO: when we have a prototype, link to/show an example of this in action.
+
+#### Methods
+
+The `.show(options)` function shows a toast element,
+by toggling its `open=""` attribute to true.
+The `options` include:
+
+- `duration`: how long to show the toast, in milliseconds. Defaults to ???
+- `multiple`: TODO
+- `newestOnTop`: TODO
+
+TODO: how do `multiple` and `newestOnTop` work?
+Should those be a global setting?
+Per container?
+How do we deal with different toasts having different values?
+
+#### Events
+
+A `<std-toast>` element can fire the following events:
+
+- `"open"`: the toast was shown
+- `"close"`: the toast was closed, either explicitly by the user, or via the timeout.
+  (Note: if animations were applied, the toast may not be entirely invisible at the time this event fires)
+    - TODO: should we consider separate events for the start and end of any close animation?
+      This seems hard to do correctly if the user customizes the animation, though.
+- `"actionclick"`: the toast's call-to-action button or link was clicked, if one exists TODO
```

Comment:
What is the benefit of including this as part of the toast API surface?

Reply:
Fair question. I think the TODO example would make this clearer. The idea would be that you could do a simple:

```js
const toast = showToast("Message", { action: "Click me" });
toast.onactionclick = () => { ... };
```

instead of something like

```js
const toast = showToast("Message", { action: "Click me" });
toast.querySelector("button:last-of-type").onclick = () => { ... };
```

which makes the assumption that the action is provided by the last button in the toast. (Which might need to be adjusted if there's a close button? Although maybe it'd be fine if the close button was in the shadow DOM??)

The reason we didn't write up this example yet is because the `{ action: "Click me" }` syntax is not settled (see above thread with @fergald).

We should probably mark this as more clearly a tentative idea.

Reply:
Ah, I missed that there's an action argument to showToast. But, my same question applies to that. I see that you're already discussing exactly this with fergal above though, so won't duplicate here.

Reply:
#11